helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java

```diff
@@ -322,6 +322,13 @@ private void computeWagedAssignmentResult(List<IdealState> wagedResourceIdealSta
       InputFields inputFields, ClusterState clusterState, String clusterId,
       AssignmentResult result) {
 
+    // If the cluster is in M mode, throw an exception
```

Comment:
nit, M mode is not official and may be misunderstood. Please use full name.

Reply:
TFTR. Updated.